### ISSUE TITLE
Corpus-based vocabulary seeding for Whisper decoding

### ISSUE THREAD

**Issue by jbarlas** [jbarlas/vox#21]
## Context

Vox currently has no way to bias Whisper's transcription toward domain-specific proper nouns and jargon (project names, contacts, technical terms) that generic Whisper models have never seen. This is a major source of real transcription errors for our actual usage (coding/agent-orchestration dictation, business/project jargon).

We considered mining "corrections" from Vox's own output history to build this vocab list, but rejected that approach: without a ground-truth signal, using Vox's own frequent outputs as training data risks reinforcing consistent misrecognitions rather than fixing them (a self-reinforcing error loop). Instead, this feature seeds the vocab list from **external, already-correct text** — sources whose content is correct by construction, not inferred from Vox's own noisy transcriptions.

## Goal

A CLI command that points Vox at a folder and/or file(s), recursively scans the textual content, extracts a ranked candidate vocabulary list (proper nouns, rare/domain-specific terms), and persists it as a local artifact that gets referenced at transcription time (Whisper `initial_prompt` biasing and/or LLM cleanup context) to improve accuracy — with minimal added latency.

## Proposed CLI surface

```
vox vocab seed <path> [<path> ...]   # recursively scan folder(s)/file(s), extract candidate terms
vox vocab list                        # show current seeded vocab + weights + source
vox vocab remove <term>               # manually exclude a term
vox vocab refresh                     # re-run extraction against previously-seeded paths
```

`vox vocab seed` should support common text-bearing formats at minimum: `.md`, `.txt`. Stretch: `.docx`/`.pdf` extraction if we already have those readers elsewhere in the codebase; source code files are out of scope for v1 (too much noise from syntax/identifiers) unless explicitly requested later.

## Extraction approach (must run fully local, no network calls)

1. Recurse through the given path(s), read all text content.
2. Tokenize and compute term frequency within the corpus, plus a rarity signal relative to a general-English reference frequency list (i.e., surface terms that are **used more than average within this corpus but rare in general English** — this is the "distinctive vocabulary" signal, not just raw frequency). A standard TF-IDF-style or log-odds-ratio-against-a-reference-corpus approach is acceptable; a bundled small reference frequency table (e.g., a common word-frequency list) should ship with Vox for the "general English" baseline so this stays fully offline.
3. Filter out common stopwords and very short tokens.
4. Rank candidates by the distinctiveness score; cap the list at a reasonable size (configurable, default ~200 terms) to avoid bloating the prompt/context.
5. Preserve original casing as it appears in the corpus (important for proper nouns and acronyms).

## How it's used at inference time

- Seeded terms should be merged into Whisper's `initial_prompt` biasing list.
- Where the LLM cleanup pass exists, seeded terms should also be available as context for that step.
- **Weighting**: corpus-seeded terms must be weighted **lower** than explicitly user-added vocab entries (i.e., a manually-added `vox vocab add <term>` entry should always take precedence / be trusted more than something inferred from a corpus scan). If both a user-added list and a corpus-seeded list exist, they should merge, not conflict, with user-added entries always winning on any collision.
- Must not meaningfully increase per-transcription latency — this is a background/offline indexing step (`vox vocab seed`/`refresh`), not something recomputed on every dictation. The resulting vocab list should be cached to disk and simply loaded at inference time.

## Acceptance criteria

- [ ] `vox vocab seed <path>` recursively scans the given folder/file(s) and produces a ranked candidate vocab list, stored locally (e.g. `~/.vox/vocab/corpus.json` or similar), tagged with its source path and a weight lower than user-added vocab.
- [ ] `vox vocab list` shows the merged effective vocab list with source (user vs. corpus) and weight, clearly distinguishing the two.
- [ ] Seeded vocab is actually consumed by the transcription pipeline (Whisper `initial_prompt` and/or LLM cleanup context) without materially increasing latency of a single dictation.
- [ ] Runs fully offline/local — no network calls during extraction or inference.
- [ ] Reasonable behavior on large corpora (e.g., a full Obsidian vault or a big folder of markdown notes) — should not hang or take an unreasonable amount of time; document expected runtime.
- [ ] Unit tests for the extraction/ranking logic using a small synthetic corpus with known distinctive terms, asserting they surface above common words.

## Out of scope for this issue

- Any correction-capture / ground-truth-from-user-edits mechanism (tracked separately).
- Fine-tuning the Whisper model itself.
- Non-text corpus sources (contacts, calendar, git repo names) — worth a fast-follow issue, but this issue is scoped to pointing at folders/files of text.

## Open questions for implementer (flag in PR if unresolved)

- Exact ranking formula (TF-IDF vs. log-likelihood-ratio vs. something simpler) — pick something simple and defensible for v1, don't over-engineer.
- Where the bundled "general English" reference frequency table comes from (a small, permissively-licensed word-frequency list is fine).
- Exact on-disk format/location for the cached vocab list — follow existing Vox config conventions if there's a precedent in the repo.
